pkg/upgrade/upgrade.go

```diff
@@ -13,12 +14,27 @@ import (
 // ManagedInstances finds all the Jaeger instances for the current operator and upgrades them, if necessary
 func ManagedInstances(c client.Client) error {
 	list := &v1.JaegerList{}
-	opts := &client.ListOptions{}
+	opts := client.MatchingLabels(map[string]string{
+		v1.LabelManagedBy: viper.GetString(v1.ConfigIdentity),
+	})
 	if err := c.List(context.Background(), opts, list); err != nil {
 		return err
 	}
 
 	for _, j := range list.Items {
+		// this check shouldn't have been necessary, as I'd expect the list of items to come filtered out already
+		// but apparently, at least the fake client used in the unit tests doesn't filter it out... so, let's double-check
+		// that we indeed own the item
+		identity := viper.GetString(v1.ConfigIdentity)
```

Comment:
This should be moved up above `opts := ...` and can then be used in the `MatchingLabels` map also.